notifications/src/main/java/org/openedx/notifications/presentation/settings/NotificationsSettingsViewModel.kt

```diff
@@ -23,39 +32,72 @@ class NotificationsSettingsViewModel(
     val notificationsConfiguration: StateFlow<NotificationsConfiguration>
         get() = _notificationsConfiguration
 
+    private val _showPermissionRequestDialog = MutableStateFlow(false)
+    val showPermissionRequestDialog: StateFlow<Boolean>
+        get() = _showPermissionRequestDialog
+
+    private val _uiMessage = MutableSharedFlow<UIMessage>()
+    val uiMessage = _uiMessage.asSharedFlow()
+
     init {
-        viewModelScope.launch {
+        if (checkPushNotificationPermission()) {
             fetchAndUpdateNotificationsSettings()
+        } else {
+            updateDiscussionPreference(updatedValue = false)
+            _showPermissionRequestDialog.update { true }
         }
     }
 
     fun setDiscussionNotificationPreference(value: Boolean) {
-        viewModelScope.launch {
-            try {
-                val response = interactor.updateNotificationsConfiguration(value)
-                updateDiscussionPreference(updatedValue = response.updatedValue)
+        if (checkPushNotificationPermission()) {
+            viewModelScope.launch {
+                try {
+                    val response = interactor.updateNotificationsConfiguration(value)
+                    updateDiscussionPreference(updatedValue = response.updatedValue)
 
-                logDiscussionPermissionToggleEvent(isDiscussionPushEnabled = value)
-            } catch (e: Exception) {
-                e.printStackTrace()
+                    logDiscussionPermissionToggleEvent(isDiscussionPushEnabled = value)
+                } catch (e: Exception) {
+                    showErrorMessage()
+                }
             }
+        } else {
+            _showPermissionRequestDialog.update { true }
         }
     }
 
-    private suspend fun fetchAndUpdateNotificationsSettings() {
-        try {
-            val response = interactor.fetchNotificationsConfiguration()
-            updateDiscussionPreference(updatedValue = response.discussionsPushEnabled)
-        } catch (e: Exception) {
-            e.printStackTrace()
+    fun fetchAndUpdateNotificationsSettings() {
+        viewModelScope.launch {
+            try {
+                val response = interactor.fetchNotificationsConfiguration()
+                updateDiscussionPreference(updatedValue = response.discussionsPushEnabled)
+            } catch (e: Exception) {
+                showErrorMessage()
+            }
         }
     }
 
-    private fun updateDiscussionPreference(updatedValue: Boolean) {
+    fun updateDiscussionPreference(updatedValue: Boolean) {
         _notificationsConfiguration.update { it.copy(discussionsPushEnabled = updatedValue) }
         preference.notifications = _notificationsConfiguration.value
     }
 
+    private fun checkPushNotificationPermission(): Boolean {
```

Comment:
better to rename the method like `isPushNotificationPermissionGranted`, cuz returning the boolean value.

Reply:
updated with `hasPushNotificationPermission`.